pkgs/applications/networking/instant-messengers/cinny-desktop/linux.nix

```diff
@@ -0,0 +1,48 @@
+{ pname
+, version
+, src
+, meta
+, stdenvNoCC
+, autoPatchelfHook
+, makeWrapper
+, wrapGAppsHook
+, dpkg
+, glib-networking
+, openssl
+, webkitgtk
+}:
+
+stdenvNoCC.mkDerivation {
+  inherit pname version src meta;
+
+  nativeBuildInputs = [
+    wrapGAppsHook
+    makeWrapper
+    autoPatchelfHook
+    dpkg
+  ];
+
+  buildInputs = [
+    glib-networking
+    openssl
+    webkitgtk
+  ];
+
+  unpackCmd = "dpkg-deb -x $src source";
+  dontConfigure = true;
+  dontBuild = true;
+  dontWrapGApps = true;
+
+  installPhase = ''
+    runHook preInstall
+    cp -r usr $out
+
+    # Environment variable fixes a blank window on nvidia:
```

Comment:
that fix was removed from the initial PR as it was deemed the place to set the environment variable was in user configuration since it depends on user setup (nvidia or not) and not in package

Reply:
Is it something that has bad side effects for people not on nvidia gpus? if not, this makes for a pretty bad user experience.

The end user expects their package to just work, they should not have to dig through upstream issues to find an environment variable that we can easily provide for them.

Reply:
I'm not sure if there are any adverse effects for nvidia users but i don't know the implications of disabling composition mode for non nvidia users either (maybe worse performance or anything, i don't know). What I did in the initial PR was to make this override conditional in the package inputs but even that was rejected when reviewed so I don't think we should add this back.

Reply:
Would you mind pinging the people who originally rejected it? I really think this is a valuable thing to add, and would love to know why they're against it.


We also have a ton of wrappers setting environment variables for stuff like wayland support, I think I've seen nvidia specifically before as well. 

You can probably also check from the wrapper what display driver is being used, and only set the variable if its nvidia. Then we fix the issue that is a significant margin of our userbase having to look through upstream issues and create their own wrappers, while we dont influence the people this doesn't effect.

Reply:
The discussion was held here https://github.com/NixOS/nixpkgs/pull/185427#discussion_r939552672

Reply:
Do you have anything specific in mind regarding detecting that a user is using nvidia proprietary drivers during the build ? (as we might not be using NixOS), seems a bit against nix way of doing things.

Reply:
> Do you have anything specific in mind regarding detecting that a user is using nvidia proprietary drivers during the build ? (as we might not be using NixOS)

We would check `/run/opengl-driver/...` to make sure the end user is on NixOS, or read `/proc/driver/nvidia/version`. Maybe `__GLX_VENDOR_LIBRARY_NAME` could also be used, although I'm not sure that's set in all desktop environments.

> seems a bit against nix way of doing things.

I would disagree. Environment-specific workarounds are not great, but as discussed earlier this is also commonly done for things like wayland compatibility. I don't see how an nvidia workaround is different. If we can ensure a package will work on all systems we deploy it to, in my opinion we should.

@SuperSandro2000 what is your point of view on this? Does not setting the environment variable on non-nvidia systems change your opinion?

Reply:
I think we can just drop that modification and use another PR for it so that we can push this one further without discussions unrelated to darwin support which is the goal of the current PR.